app/assets/javascripts/tags.js.coffee

```diff
@@ -1,17 +1,34 @@
-$(document).on "click", ".switch_offer", (event) ->
-  $.ajax '/tags/offers',
-    type: 'GET'
-    dataType: 'html'
-    error: (jqXHR, textStatus, errorThrown) ->
-        $('.alpha_tag_list').html "AJAX Error: #{textStatus}"
+$ ->
+  $(".switch_offer-js").on "click", ->
```

Comment:
Even if in `$(document).ready()` callback, it's better to use the extended `$(document).on("event", "selector", callback)` so new elements created programmatically can be listened for events. Otherwise only elements that are matched by the initial selector are listened for events.